implementation/README.md

```diff
@@ -0,0 +1,147 @@
+1.  Replace UEFI Shell code section with Linux kernel (change part of one thing)
+1.  Remove as many DXEs as possible (change by removal). This change:
+
+       *  Speeds boot
+       *  Reduces panic possibilities
+       *  Removes exploits
+       *  In production, it has solved problems
+
+1.  Replace some DXEs with open source components (change by replacement)
+
+## Tools of the trade
+
+There are two tools you use when you modify the UEFI flash image: `utk `and `me_cleaner`
+
+The `utk` tool can:
+
++   Remove DXEs
++   Insert new DXEs
++   Replace the binary code of a DXE with a kernel
+
+If you are lucky, you can always flash with _internal_ flash hardware.
+
+The Management Engine (ME) is an x86 CPU embedded in the Intel Platform Controller Hub (PCH). It runs the Minix operating system which boots first and enables hardware such as clocks and GPIOs. ME checks the contents of flash memory and is used to implement "BootGuard". If you reflash and the ME is in "BootGuard" mode, your machine will be unusable. You need to run a tool called `me_cleaner` on the image to disable BootGuard.
+
+     /usr/bin/python2 me_cleaner.py -s somefile.bin
+
+`me_cleaner` sets the high assurance platform (HAP) bit. HAP is an IT security framework that provides a way to disable a feature on Intel chips that presents a security risk. Setting the bit with `me_cleaner` disables the "feature".  Note that this does not always work; check with your hardware vendor.  
+`
+When you run `me_cleaner`
+
+    ~/projects/linuxboot/me_cleaner/me_cleaner.py -s /tmp/rom.bin
+
+you should see output similar to the following:
+
+
+|  |
+|:---|
+|`Full image detected`|
+|`Found FPT header at 0x1010`|
+|`Found 20 partition(s)`|
+|`Found FTPR header: FTPR partition spans from 0x6f000 to 0xe700`|
+|`ME/TXE firmware version 2.0.5.3112 (generation 2)`|
+|`Public key match: Intel TXE, firmware versions 2.x.x.x`|
+|`The AltMeDisable bit is SET`|
+|`Setting the AltMeDisable bit in PCHSTRP10 to disable Intel ME…`|
+|`Checking the FTPR RSA signature... VALID`|
+|`Done! Good luck!`|
+
+## LinuxBoot Implementation steps
+
+### Step 1: boot Linux via netboot / UEFI shell
+
++   netboot: standard BIOS-based PXE boot
+
+    +   Netboot is probably the most common working boot method on UEFI
+    +   We have never seen a system that did not have a net boot
+
++   UEFI Shell (mentioned only for completeness)
+
+    +   Install Linux on FAT-32 media with a name of your choice (e.g. "kernel")
+        +   FAT-32, also known as MS-DOS file system
+    +   Boot kernel at UEFI Shell prompt
+    +   We've run into a few systems that don't have a UEFI shell
+
+#### Working with a system that only has a net interface
+
+If the system only has a net interface, you use Dynamic Host Configuration Protocol (DHCP), using broadcast DISCOVER, and Trivial File Transfer Protocol (TFTP) to get the boot information you need. These programs are provided by the PXEboot program.
```

Comment:
These programs ... 
is borderline not quite right, I think we can remove this sentence.

Reply:
I removed the sentence.